src/social-network-provider/twitter.com/ui/fetch.ts

```diff
@@ -1,102 +1,66 @@
-import {
-    bioCard,
-    fromPostSelectorsSelectPostContentString,
-    postsRootSelector,
-    postsSelectors,
-    selfInfoSelectors,
-} from '../utils/selector'
+import { bioCard, postParser, postsRootSelector, postsSelectors, selfInfoSelectors } from '../utils/selector'
 import { MutationObserverWatcher } from '@holoflows/kit'
-import Services from '../../../extension/service'
 import { PersonIdentifier } from '../../../database/type'
 import { host } from '../index'
 import { getEmptyPostInfo, SocialNetworkUI, SocialNetworkUIInformationCollector } from '../../../social-network/ui'
 import { deconstructPayload } from '../../../utils/type-transform/Payload'
-import { regexMatch } from '../../../utils/utils'
-import { equal, notEmpty } from '../../../utils/assert'
+import { notEmpty } from '../../../utils/assert'
 import { instanceOfTwitterUI } from './index'
 import { resolveInfoFromBioCard } from '../utils/fetch'
+import { uploadToService } from '../utils/user'
 
 const resolveLastRecognizedIdentity = (self: SocialNetworkUI) => {
     const selfSelector = selfInfoSelectors().screenName
     const assign = () => {
-        try {
-            const ref = self.lastRecognizedIdentity
-            const info = selfInfoSelectors().screenName.evaluate()
-            const id = new PersonIdentifier(host, notEmpty(info, 'user id not found')[0])
-            equal(id.isUnknown, false, 'user id not recognized')
-            ref.value = { identifier: id }
-        } catch (e) {
-            /**
-             * TODO:
-             *  This function expects to fail to collect info,
-             *  maybe don't throw is a good idea
-             *                                  -- Jack-Works
-             */
-            console.log(e)
+        const ref = self.lastRecognizedIdentity
+        const screenName = selfInfoSelectors().screenName.evaluate()
+        const nickname = selfInfoSelectors().name.evaluate()
+        const avatar = selfInfoSelectors().userAvatar.evaluate()
+        const identifier = new PersonIdentifier(host, notEmpty(screenName, 'user id not found'))
```

Comment:
use `PersonIdentifier.unknown` if you did not find anything.

Reply:
c7a01283